documentation/add/04_mpi_framework.md

```diff
@@ -1,43 +1,244 @@
+| `cluster` | P | Exposes read access to the class, controlling cluster, i.e. the pool of independent processes or programs, to execute parallel job. 
+| `is_initialized` | P | true if *JobDispatcher* already controls a cluster so the next job can be executed on existing cluster rather then after starting a new one. False if the cluster is not running and needs to be started up |
+|**Job control properties**: | :: | |
+| `task_check_time` | P |  how often (in second) job dispatcher should query the task status and report progress to user |
+| `fail_limit` | P | number of times to try action until deciding the action have failed |
+| `time_to_fail` | P | Time interval to wait until job which do not return any messages from the cluster is considered failed and should be terminated. |
+|**Job control Methods** : | :: | |
+| `split_tasks` | M | Auxiliary method, used internally by the following methods and taking as input the structure, containing the information common to all parallel workers, number of workers and cellarray of input parameters to split among the workers. Returns the array of messages, to initialize each worker of the cluster. | 
+| `start_job` | M | Taking as input the number of workers requested, job name and its input job parameters, splits the job among workers using *split\_tasks*, starts the cluster and controls the job execution, regularly querying the *cluster* on the progress of the job execution and displaying the results of the execution. Returns the result of the job. |
+| `restart_job` | M |  The same as *start_job* but does not start the cluster using existing cluster to start new or continue the old job providing new input data. |
+
+One of the main properties of *JobDispatcher* class is *cluster* property, containing an instance of *ClusterWrapper* class, responsible for control of the pool of multiple parallel processes or programs, i.e. the *cluster*. The *JobDispatcher* controls and communicates with the cluster through the properties, described below. 
+
+
+
+## Cluster Management 
+Cluster classes are responsible for launching and initializing parallel workers, displaying job progress and completing parallel execution. The parent for all cluster classes is a *ClusterWrapper* class, which provides common interface for the communicators/controllers managing the parallel processes itself. **Fig 3** shows *ClusterWrapper* interface and lists the classes, perform its actual software implementations. 
+
+![Fig 3: Cluster Wrapper](../diagrams/ClusterWrapperInterface.png)
+
+**Fig 3** Cluster Wrappers interface and its current implementations. 
+
+To provide simple selection of a framework, all Cluster Wrapper classes are subscribed to framework factory **MPI\_fmwks\_factory**. User interacts with the factory through **parallel\_config** configurations class:
```

Comment:
I changed it to be more vague, but framework and cluster are almost the same now. May be the name should change -- let's not do it now. 
I've added small description of the framework factory though not that sure its useful. Its not fully developing framework code, left for expansion in the future when we have different cluster configurations.

Reply:
Probably worth to make separate ticket as its indeed clusters factory rather than frameworks factory. https://github.com/pace-neutrons/Herbert/issues/162